### ISSUE TITLE
[g-image] After build I have image with 3x of the original size

### ISSUE THREAD

**Issue by Kulcanhez** [gridsome/gridsome#1256]
## Description

g-image is compressing the image 3x of the original size of the image.

### Steps to reproduce

1. Use a compressed image in a g-image (Use an ultra compressed image for easier testing. You can use [ImageOptim](https://imageoptim.com/) with the JPEG quality in the minimum).
2. Compile using `gridsome build`
3. Check the size of the image before and after

### Expected result

Should be an option to bypass the compression at least for the original/main resolution of the image and only compress for other sizes but respecting the `quality` parameter. Or check if the size of the compressed image is higher and bypass the compression. @hjvedvik tagging you 'cause you're assigned to all tasks related to g-image ❤️

_Edited:_
**Three ways I see this can be done for the developer:**
1. Option to disable compression
2. Check if the compressed file is bigger than the original and ignore it
3. `quality="100"` means no compression

### Actual result

The original size is 92kb and after the build I have 267kb.

![](https://user-images.githubusercontent.com/5109593/86495209-4338bc80-bd70-11ea-87ff-0d706e9a6da8.jpg)

### Environment

```
  System:
    OS: macOS 10.15.5
    CPU: (4) x64 Intel(R) Core(TM) i5-5257U CPU @ 2.70GHz
  Binaries:
    Node: 12.16.3 - /usr/local/bin/node
    npm: 6.14.5 - /usr/local/bin/npm
  Browsers:
    Safari: 13.1.1
  npmPackages:
    @gridsome/plugin-sitemap: 0.4.0 => 0.4.0 
    @gridsome/source-filesystem: 0.6.2 => 0.6.2 
    @gridsome/transformer-remark: 0.6.1 => 0.6.1 
    gridsome: 0.7.18 => 0.7.18 
    gridsome-plugin-robots: 0.2.1 => 0.2.1 
  npmGlobalPackages:
    @gridsome/cli: 0.3.3
```

**Comment by noxify** [comment]
Related https://github.com/gridsome/gridsome/issues/1091

**Comment by Kulcanhez** [comment]
4 slider images before building:
![](https://user-images.githubusercontent.com/5109593/86510724-1aeda400-bdea-11ea-962c-1c59e54cd349.jpg)

After building:
![](https://user-images.githubusercontent.com/5109593/86510741-33f65500-bdea-11ea-9dea-327fa77a3f1c.jpg)

I'm using: 
`<g-image src="~/../uploads/1.jpg" quality="100"/>`

**Three ways I see this can be done for the developer:**
1. Option to disable compression
2. Check if the compressed file is bigger than the original and ignore it
3. `quality="100"` means no compression

@hjvedvik but you will be the judge for the best way to achieve it 😄

**Comment by hjvedvik** [comment]
I did some more work on this tonight and think I have a solution that works well now. But I have to wait until tomorrow before publishing a new release. I basically just implemented all your suggestions, @Kulcanhez :) They seem to work well as far as I've tested.

It will be possible to disable processing for all images by setting `images: { minimize: false }` in `gridsome.config.js` after the release. And for a single image with `quality="100"`.  Images are still resized to the different sizes when processing is disabled as long as the resulting file size is not exceeding the initial size.

**Comment by hjvedvik** [comment]
@Kulcanhez Just published v0.7.19 :) I named the option `compress`, which is `true` by default. And I also added `defaultQuality: 75`. Setting `defaultQuality: 100` is basically the same as `compress: false`.

```js
module.exports = {
  images: {
    compress: false // disables image compression
  }
}
```

**Comment by Kulcanhez** [comment]
> @Kulcanhez Just published v0.7.19 :) I named the option `compress`, which is `true` by default. And I also added `defaultQuality: 75`. Setting `defaultQuality: 100` is basically the same as `compress: false`.
> 
> ```js
> module.exports = {
>   images: {
>     compress: false // disables image compression
>   }
> }
> ```

@hjvedvik I tested yesterday and is working amazingly!! Really thank you for the fast fix and for being open to suggestions ❤️ 🚀

**Comment by perpetualgrimace** [comment]
Just wanted to pop in and say I'm not sure these settings are working for JPGs. I tried setting compress to false, and tried different defaultQuality settings, and also tried the quality attribute in the g-image, none seemed to have an effect.
